pkg/service/form_gallery.go

```diff
@@ -241,28 +241,25 @@ func GetFormGallery(c *web.AppContext) (model.FormGallery, *web.AppError) {
 
 func UpdateFormGallery(context *web.AppContext) (*model.FormGallery, *web.AppError) {
 
-	idStr := context.GetValue("gallery_id")
-	id := bson.ObjectIdHex(idStr)
+	// unmarshall the input
+	var input model.FormGallery
+	if err := UnmarshallAndValidate(context, &input); err != nil {
+		return nil, err
+	}
 
 	// ensure the form gallery exists
 	var dbEntity model.FormGallery
-	err := context.MDB.DB.C(model.FormGalleries).FindId(id).One(&dbEntity)
+	err := context.MDB.DB.C(model.FormGalleries).FindId(input.ID).One(&dbEntity)
```

Comment:
Your technique of updating based on the id in the json is better than relying on the url string.  If we rely on the url string it is possible that differing ids can be sent, which would end up in a gallery changing ids.  This should never be possible.

Good call.